BedrockServer.cpp

```diff
@@ -2148,54 +2148,38 @@ void BedrockServer::handleSocket(Socket* s) {
         // are in a `shutting down` state, then we finish up and exit. In any other case, we just wait in `poll` again
         // until we get some data or a disconnection.
         int pollResult = 0;
-        bool exitMainLoop = false;
         struct pollfd pollStruct = { s->s, POLLIN, 0 };
 
-        // This loop just calls `poll` sequentially optionally exiting if we're shutting down and haven't received any
-        // data.
-        while (true) {
-            // Do the actual `poll` call.
-            pollResult = poll(&pollStruct, 1, 1'000);
+        // As long as `poll` returns 0 we've timed out, indicating that we're still waiting for something to happen. In
+        // that case, we'll loop again *unless* we're shutting down.
+        while (!(pollResult = poll(&pollStruct, 1, 1'000))) {
+            if (_shutdownState != RUNNING) {
+                SINFO("Socket thread exiting because no data and shutting down.");
+                s->state = Socket::CLOSED;
+                ::shutdown(s->s, SHUT_RDWR);
```

Comment:
Just jotting down what I believe I understand here since I couldn't find a `shutdown` method defined in bedrock:

`::` is saying to look in the global scope for the the method `shutdown`

I believe then this is calling linux's built in `shutdown` method https://man7.org/linux/man-pages/man2/shutdown.2.html? 

I see we have a method `STCPManager::shutdownSocket` that does some similar stuff though the states are different but seems to check out that this is a good way of closing a socket and signalling to shut down.

Reply:
Yes.